saleor/discount/utils.py

```diff
@@ -93,7 +96,8 @@ def calculate_discounted_price(
     price: Money,
     collections: Iterable["Collection"],
     discounts: Optional[Iterable[DiscountInfo]],
-    channel: "Channel"
+    channel: "Channel",
+    variant_id: Optional[int] = None
```

Comment:
Why `variant_id` is optional here and `product` is required?

Reply:
Mainly, because `def update_product_discounted_price(product, discounts=None):` in `variant_prices.py` also uses this function, and there, passing a variant_id will be somewhat artificial (it would be even hard to get that kind of data, because of how `variant_prices_dict` looks like.